File: Kotlin/tests/AntTest.kt
class AntTest() {

    val position = Position(0,0)

    fun run() {
        canTurn()
        canMoveForward()
    }

    fun canTurn(){
        val ant = Ant(position, Direction(0,1));
        ant.turn(Turn.Left);
        assert(ant.direction==Direction(-1,0), "ant should turn left")
    }

    fun canMoveForward(){
        val ant = Ant(Position(1,2), Direction(3,4));
        ant.move();
        assert(ant.position==Position(4, 6), "ant should move to another position");
    }

}

File: CSharp/Langton.Tests/AntTest.cs
using FluentAssertions;
using NUnit.Framework;

namespace Langton.Tests
{
	public class AntTest
	{
		readonly Position _initialPosition = new Position(0, 0);
		readonly Direction _initialDirection = Direction.North;

		[TestCase(Rotation.Left)]
		[TestCase(Rotation.Right)]
		public void CanTurn(Rotation rotation)
		{
			var ant = new Ant(_initialDirection, _initialPosition);
			var expectedDirection = _initialDirection.Rotate(rotation);
			ant.Turn(rotation);
			ant.Direction.Should().Be(expectedDirection);
		}

		[Test]
		public void CanMoveForward()
		{
			var ant = new Ant(_initialDirection, _initialPosition);
			ant.MoveForward();
			ant.Position.Should().Be(_initialDirection.MoveFrom(_initialPosition));
		}
	}
}